tmt/base.py

```diff
@@ -104,12 +104,13 @@ class FmfId(
         tmt.export.Exportable['FmfId']):
 
     # The list of valid fmf id keys
-    VALID_KEYS: ClassVar[List[str]] = ['url', 'ref', 'path', 'name']
+    VALID_KEYS: ClassVar[List[str]] = ['url', 'ref', 'path', 'name', '_ref_expanded']
```

Comment:
Hm, a sudden realization: we need to handle `_ref_expanded` in a different way, as it's not the same kind of key as `ref` or `path`. The main reason is, `_ref_expanded` is not a key fmf id specification defines, yet it appears to be. It's not allowed to use this key in `require`, for example, the key should reflect that.

* it shouldn't be in this list, as it's not a valid key, recognized on input. Its presence allows `_ref_expanded` to be used in `link` specification, for example, which is not correct.
* it should be forcefully removed in `to_spec()` and `to_minimal_spec()` methods, it's not a key one could get from tmt and feed it back as part of fmf id specification.
* I'm not 100 % sure, but ^^ seems to apply to `export()` as well, same reason.

It's already sneaking out of its box, `if key == '_ref_expanded'` when logging fmf id components should not be a thing, `_ref_expanded` should not appear in the minimal spec, and code should not be ready to skip it.

Reply:
@happz I am not sure I did it right, could you please check it? Also, I think that `export()` modifications should be necessary, it seems that attributes prefixed with `_` are skipped.

Reply:
> @happz I am not sure I did it right, could you please check it?

Looks good to me. One left over though, there should be no need for `if key == '_ref_expanded'` checks anymore as the key should be now gone from mappings returned by `to_spec()` and `to_minimal_spec()` calls.

> Also, I think that `export()` modifications should be necessary, it seems that attributes prefixed with `_` are skipped.

Correct, I forgot we do that already.

Reply:
This should be hopefully addressed now.

Reply:
One leftover, one comment, that should be all. I see no other issue.